PresentData/PresentMonTraceConsumer.cpp

```diff
@@ -271,6 +281,10 @@ void PMTraceConsumer::HandleDxgkFlip(EVENT_HEADER const& hdr, int32_t flipInterv
 
     // If this is the DWM thread, piggyback these pending presents on our fullscreen present
     if (hdr.ThreadId == DwmPresentThreadId) {
+        for (auto iter = mPresentsWaitingForDWM.begin(); iter != mPresentsWaitingForDWM.end(); iter++)
+        {
+            iter->get()->PresentInDwmWaitingStruct = false;

```

Comment:
Note that this optimization was not present in the first iteration. We now only check mPresentsWaitingForDWM in RemoveLostPresents if the list hasn't been swapped out.